air-script/tests/aux_trace/aux_trace.masm

```diff
@@ -1,3 +1,22 @@
+proc.cache_z_exp
+    padw mem_loadw.4294903304 drop drop
+    # => [z_1, z_0, ...]
+    # Exponentiate z trace_len times
+    mem_load.4294903307 neg dup.0 neq.0
+    while.true
+        movdn.2 dup.1 dup.1 ext2mul
+        # => [z_1^2, z_0^2, i, ...]
```

Comment:
minor nit on notation: technically, the values on the stack are not `[z_1^2, z_0^2, ...]`, but more like `[(z_0, z_1)^2, ...]`.

Reply:
minor nit: the same is done [here](https://github.com/0xPolygonMiden/miden-vm/blob/4576337fbc8702364e01d1a505453f033048a5eb/stdlib/asm/crypto/stark/random_coin.masm#L422), so we should probably leave a TODO to avoid duplication

Reply:
> minor nit on notation

changed

> the same is done [here](https://github.com/0xPolygonMiden/miden-vm/blob/4576337fbc8702364e01d1a505453f033048a5eb/stdlib/asm/crypto/stark/random_coin.masm#L422)

this is slightly different. Depending on the Air definition this will compute intermediary values used to evaluate the periodic column polynomial. Here is an [example](https://github.com/0xPolygonMiden/air-script/blob/714fe2f5bbccddcc1cfad861858c788253bbeb82/air-script/tests/bitwise/bitwise.masm#L4-L15).

I created an issue on the miden-vm repo https://github.com/0xPolygonMiden/miden-vm/issues/952